### PR TITLE
:sparkles: Aws iam auth api change request

### PR DESCRIPTION
**Pull request by suvaanshkumar**
## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by qiujian16** [comment]
you would need to update the types.go at first, and then run make update to generate crd and client code. Ensure to run make verify after make update. Here https://github.com/open-cluster-management-io/api/blob/main/docs/development.md has some infos.

**Comment by EricaJ6** [comment]
Thanks @qiujian16 for the suggestions. Please take a look at the updated changes with required changes.

File: operator/v1/types_klusterlet.go [hunk 1]
```diff
@@ -172,6 +172,28 @@ type RegistrationConfiguration struct {
...
+
+type RegistrationDriver struct {
+	// Type of the authentication used by managedcluster to register as well as pull work from hub. Possible values are csr and awsirsa.
+	// +kubebuilder:default:=csr
+	// +kubebuilder:validation:Enum=csr;awsirsa
+	AuthType string `json:"authType"`
+
+	// Contain the details required for registering with eks hub cluster using IAM roles for service account.
+	// This is required only when the authType is awsirsa.
+	AwsIrsa AwsIrsa `json:"awsIrsa,omitempty"`
```
Comment (qiujian16) [inline operator/v1/types_klusterlet.go]: this should be the pointer

File: operator/v1/types_klusterlet.go [hunk 2]
```diff
@@ -172,6 +172,28 @@ type RegistrationConfiguration struct {
...
+}
+
+type AwsIrsa struct {
+	// The arn of the EKS hub cluster. This will be required to pass information to hub, which hub will use to create IAM identities for this klusterlet.
+	// Example - arn:eks:us-west-2:12345678910:cluster/hub-cluster1.
+	// +required
+	EksHubClusterArn string `json:"eksHubClusterArn"`
```
Comment (qiujian16) [inline operator/v1/types_klusterlet.go]: maybe just hubClusterArn

File: operator/v1/types_klusterlet.go [hunk 3]
```diff
@@ -172,6 +172,28 @@ type RegistrationConfiguration struct {
...
+	// +kubebuilder:validation:Enum=csr;awsirsa
+	AuthType string `json:"authType"`
+
+	// Contain the details required for registering with eks hub cluster using IAM roles for service account.
+	// This is required only when the authType is awsirsa.
+	AwsIrsa *AwsIrsa `json:"awsIrsa,omitempty"`
+}
+
+type AwsIrsa struct {
+	// The arn of the EKS hub cluster. This will be required to pass information to hub, which hub will use to create IAM identities for this klusterlet.
```
Comment (mikeshng) [inline operator/v1/types_klusterlet.go]: @qiujian16 do we still need to mention EKS here specifically after your field rename suggestion? Should we do something like: "The arn of the hub cluster (ie: an EKS cluster)."

File: operator/v1/types_klusterlet.go [hunk 4]
```diff
@@ -172,6 +172,28 @@ type RegistrationConfiguration struct {
...
+	RegistrationDriver RegistrationDriver `json:"registrationDriver,omitempty"`
+}
+
+type RegistrationDriver struct {
+	// Type of the authentication used by managedcluster to register as well as pull work from hub. Possible values are csr and awsirsa.
+	// +kubebuilder:default:=csr
+	// +kubebuilder:validation:Enum=csr;awsirsa
+	AuthType string `json:"authType"`
+
+	// Contain the details required for registering with eks hub cluster using IAM roles for service account.
```
Comment (mikeshng) [inline operator/v1/types_klusterlet.go]:
```suggestion
	// Contain the details required for registering with hub cluster (ie: an EKS cluster) using AWS IAM roles for service account.
```
I think it's better to be consistent and use EKS in upper case in all the places.

File: operator/v1/types_klusterlet.go (hunk 3 above)
Comment (EricaJ6) [reply operator/v1/types_klusterlet.go]: This arn is specific to the EKS cluster only. I can update the description to `The arn of the hub cluster (ie: an EKS cluster).`

**Comment by qiujian16** [comment]
/approve

@mikeshng @elgnay would you take the review and tag?

**Comment by elgnay** [comment]
LGTM

**Pull request merged**